client.go

```diff
@@ -151,11 +152,16 @@ func newClient(
 		tlsConf = &tls.Config{}
 	}
 	if tlsConf.ServerName == "" {
-		var err error
-		tlsConf.ServerName, _, err = net.SplitHostPort(host)
-		if err != nil {
-			return nil, err
+		sni := host
+		if strings.IndexByte(sni, ':') != -1 {
```

Comment:
Why not `strings.Contains`?

Reply:
@marten-seemann 
`IndexByte` could be faster, so I prefer it to `Contains` when the target is only a single character.

Just wrote a benchmark:
```go
package main
import (
	"testing"
	"strings"
)

func BenchmarkMethod(b *testing.B) {
	var ok bool
	s := strings.Repeat("0", 10)
	for i := 0; i < b.N; i++ {
		//ok = strings.Contains(s, "a")
		ok = strings.IndexByte(s, 'a') != -1
	}
	println(ok)
}
```

`IndexByte` is 10% faster in my machine.

Reply:
Fair enough. I don't think it matters, since this is only executed a single time per connection. But if you insist, I'm ok with keeping `IndexByte`.